apm-agent-premain/src/main/java/co/elastic/apm/agent/premain/AgentMain.java

```diff
@@ -174,6 +177,43 @@ private static File getAgentJarFile() throws URISyntaxException {
         if (!agentJar.getName().endsWith(".jar")) {
             throw new IllegalStateException("Agent is not a jar file: " + agentJar);
         }
-        return agentJar.getAbsoluteFile();
+        // verify agent shading at runtime, because tests might not have been always executed (when creating plugin)
+        // and failure to properly shade will create unexpected and complex behavior that is hard to diagnose and
+        // debug, thus the cost of doing this at runtime is worth it.
```

Comment:
The problem is that when we add something intentionally to the jar, we need to maintain both here and `co.elastic.apm.servlet.AgentFileIT#testEverythingIsShaded`. I don't see how something can be merged without making sure tests pass. So the latest we should stumble into this, is when running the integration tests, or running manual tests (which is normally done after integration tests pass).
If you feel this is important, at least add a reference from each of the location to the other, so that when one is updated, the other is updated as well.

Reply:
I agree, that completely duplicates that IT test, the issue I had with that was more related to the long feedback loop that it implies:
- integration tests tend to take a while, thus I don't run them all the time
- they are finally run in CI when PR/branch is pushed, thus we are sure that we won't merge with such test failing
- the unexpected side effects of improper packaging (things not working as expected) can thus be silently ignored until IT are run

When trying to have a short feedback loop and thus skipping long-running tests (which tends to be common, at least in my current plugin dev workflow), we might spend quite some time until figuring out that there is a packaging issue.

Given this issue might definitely go away once we have proper external plugin and do not require shading anymore, it's not worth keeping that here, I have removed it and kept only the `testEverythingIsShaded` variant.